### PR TITLE
String to int encapsulation

### PR DESCRIPTION
**Pull request by ciphx**
Jira : https://jira.baeldung.com/browse/BAEL-5859
Link to article: https://drafts.baeldung.com/wp-admin/post.php?post=159699&action=edit

### CODE REVIEW

File: core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package com.baeldung.stringtoint;
+
+import java.util.Optional;
+import org.apache.commons.lang3.math.NumberUtils;
+
+public class StringToIntEncapsulation {
```
Comment (vinipx) [inline core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java:6]: _StringToIntConverter_ sounds more appropriate, no?

File: core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public static Integer convertStringToIntUsingIntegerDecode(String input){
+        try {
+            return Integer.decode(input);
+        } catch (Exception e) {
+            // log or handle the error
+            return null; // or Integer.MIN_VALUE, or some other default value
+        }
+    }
+
+    public static Integer converStringToIntUsingOptional(String input){
```
Comment (vinipx) [inline core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java:38]: conver**t**StringToIntUsingOptional

File: core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        Optional<Integer> parsedInt;
+        try {
```
Comment (vinipx) [inline core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java:40]: As per described in the video review, I'd rather remove this _try-catch_ block and go with this instead:

**return Optional.ofNullable(Integer.parseInt(s)).orElse(null);** 

(remember that **null** not necessarily is the best option here)

File: core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            parsedInt = Optional.of(Integer.parseInt(input));
+        } catch (Exception e) {
+            // log or handle the error
+            parsedInt = Optional.empty();
+        }
+        return parsedInt.orElse(null);
+    }
+
+    public static int convertStringToIntUsingNumberUtils(String input){
+        //returns Integer.MIN_VALUE as the default value if conversion fails
```
Comment (vinipx) [inline core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java:50]: I don't think this comment is really necessary - it is explicit by the line 51 itself.

File: core-java-modules/core-java-string-conversions/src/test/java/com/baeldung/datetostring/DateToStringFormatterUnitTest.java
Comment (vinipx) [inline core-java-modules/core-java-string-conversions/src/test/java/com/baeldung/datetostring/DateToStringFormatterUnitTest.java:1]: This file have nothing to do with the article - better to remove it.

**Review by vinipx — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java (hunk 3 above)
Comment (ciphx) [reply core-java-modules/core-java-string-conversions/src/main/java/com/baeldung/stringtoint/StringToIntEncapsulation.java:40]: so orElse doesn't handle the scenario when parsing fails. It throws a number format exception. It only handles scenarios where we receive a null value.

**Review by vinipx — APPROVED** [review]

**Pull request merged**